src/expr/impl/src/scalar/cmp.rs

```diff
@@ -287,6 +289,64 @@ fn is_not_null<T>(v: Option<T>) -> bool {
     v.is_some()
 }
 
+#[function("greatest(...) -> boolean")]
+#[function("greatest(...) -> *int")]
+#[function("greatest(...) -> decimal")]
+#[function("greatest(...) -> *float")]
+#[function("greatest(...) -> serial")]
+#[function("greatest(...) -> int256")]
+#[function("greatest(...) -> date")]
+#[function("greatest(...) -> time")]
+#[function("greatest(...) -> interval")]
+#[function("greatest(...) -> timestamp")]
+#[function("greatest(...) -> timestamptz")]
+#[function("greatest(...) -> varchar")]
+#[function("greatest(...) -> bytea")]
+pub fn general_variadic_greatest<T>(row: impl Row) -> Option<T>
+where
+    T: Scalar,
+    for<'a> <T as Scalar>::ScalarRefType<'a>: TryFrom<ScalarRefImpl<'a>> + Ord + Debug,
+{
+    row.iter()
+        .flatten()
+        .map(
+            |scalar| match <<T as Scalar>::ScalarRefType<'_>>::try_from(scalar) {
+                Ok(v) => v.to_owned_scalar(),
```

Comment:
It seems we can `max` on `ScalarRef` and only `to_owned_scalar` on the max found at the end.